apps/server/src/modules/oauth/controller/oauth-sso.controller.spec.ts

```diff
@@ -183,4 +184,57 @@ describe('OAuthController', () => {
 			await expect(controller.requestAuthToken(currentUser, request, '')).rejects.toThrow(UnauthorizedException);
 		});
 	});
+
+	describe('migrateUser', () => {
+		describe('when migration was successful ', () => {
+			it('should redirect to migration succeed page ', async () => {
+				const currentUser: ICurrentUser = { userId: 'userId' } as ICurrentUser;
+				const { res } = getMockRes();
+				const query: AuthorizationParams = new AuthorizationParams();
+				query.code = 'defaultAuthCode';
+				const systemParams: SystemUrlParams = new SystemUrlParams();
+				systemParams.systemId = 'systemId';
+
+				oauthUc.migrateUser.mockResolvedValue({ redirect: `${mockHost}/migration/succeed` });
+
+				await controller.migrateUser(currentUser, query, res, systemParams);
+
+				expect(res.redirect).toHaveBeenCalledWith(`${mockHost}/migration/succeed`);
+			});
+		});
+
+		describe('when migration failed ', () => {
+			it('should redirect to dashboard ', async () => {
+				const currentUser: ICurrentUser = { userId: 'userId' } as ICurrentUser;
+				const { res } = getMockRes();
+				const query: AuthorizationParams = new AuthorizationParams();
+				query.code = 'defaultAuthCode';
+				const systemParams: SystemUrlParams = new SystemUrlParams();
+				systemParams.systemId = 'systemId';
+
+				oauthUc.migrateUser.mockResolvedValue({ redirect: `${mockHost}/dashboard` });
+
+				await controller.migrateUser(currentUser, query, res, systemParams);
+
+				expect(res.redirect).toHaveBeenCalledWith(`${mockHost}/dashboard`);
+			});
+		});
+
+		describe('when migration redirect is not given ', () => {
+			it('should throw InternalServerErrorException ', async () => {
+				const currentUser: ICurrentUser = { userId: 'userId' } as ICurrentUser;
+				const { res } = getMockRes();
+				const query: AuthorizationParams = new AuthorizationParams();
+				query.code = 'defaultAuthCode';
+				const systemParams: SystemUrlParams = new SystemUrlParams();
+				systemParams.systemId = 'systemId';
+
+				oauthUc.migrateUser.mockResolvedValue(new UserMigrationDto({}));
+
+				await expect(controller.migrateUser(currentUser, query, res, systemParams)).rejects.toThrow(
+					InternalServerErrorException
```

Comment:
Think about checking a specific message, because on some other point could thrown an error like this and you dont know if it is the correct one